archeryutils/targets.py

```diff
@@ -244,32 +330,62 @@ def min_score(self) -> float:
         >>> mytarget.min_score()
         1.0
         """
-        if self.scoring_system in (
-            "5_zone",
-            "10_zone",
-            "10_zone_compound",
-            "WA_field",
-            "IFAA_field_expert",
-            "Worcester",
-        ):
-            return 1.0
-        if self.scoring_system in (
-            "10_zone_6_ring",
-            "10_zone_6_ring_compound",
-        ):
-            return 5.0
-        if self.scoring_system in (
-            "10_zone_5_ring",
-            "10_zone_5_ring_compound",
-        ):
-            return 6.0
-        if self.scoring_system in ("Worcester_2_ring",):
-            return 4.0
-        if self.scoring_system in ("IFAA_field",):
-            return 3.0
-        if self.scoring_system in ("Beiter_hit_miss"):
-            # For Beiter options are hit and miss, so return 0 here
-            return 0.0
+        if self.is_custom:
+            return min(self._face_spec.values())
+        data = _scoring_system_data.get(self.scoring_system)
+        if data:
+            return data.low
         # NB: Should be hard (but not impossible) to get here without catching earlier.
         msg = f"Target face '{self.scoring_system}' has no specified minimum score."
         raise ValueError(msg)
+
+    def get_face_spec(self) -> FaceSpec:
+        # Could replace with mapping face to lambda that returns spec
+        """Derive specifications for common/supported targets.
+
+        Returns
+        -------
+        spec : dict
+            Mapping of target ring sizes in [metres] to score
+        """
+        system = self.scoring_system
+        tar_dia = self.diameter
+
+        if system == "Custom":
+            return self._face_spec
+
+        data = _scoring_system_data.get(system)
+        if not data:
+            # no data for scoring system, raise
+            msg = f"No rule for calculating scoring for face type {system}."
+            raise ValueError(msg)
+
+        # calculate missing rings for certain targets from minimum score
+        missing = data.low - 1
+
+        if system == "5_zone":
```

Comment:
Had a look at it already and it looks slightly better as a match statement though not as big a difference as I'd thought; single dispatch isn't really using the full power of pattern matching.
I did start looking at fancier matching patterns to try and derive compound scoring/number of rings from the system name but I think it actually just obfuscates the logic further, theres already a fair amount (less than before) of magic number wrangling dispatched on which scoring system we use.

It would also be pretty trivial to do after the fact, but again if #82 goes in first I'll make the switch in this at that time.